### ISSUE TITLE
Multiplayer connection bug

### ISSUE THREAD

**Issue by L0tion** [bluemagic123/tModLoader#2036]
### Version

Latest Official

### OS

Windows

### Platform

Steam

### Description

When I try to connect to my friend's world on steam, it shows a message in the image i sent, and doesn't let me into their world.

### Log File

![image](https://user-images.githubusercontent.com/95969253/145679011-33fd2e9b-e654-4143-8fd1-ac1d5170a4f3.png)

**Comment by SimoneG97** [comment]
I suggest to specify at least if you are using 1.3 or 1.4 version, since "latest official" is kinda ambiguous
EDIT: Nevermind, my brain farted

**Comment by direwolf420** [comment]
Latest Official is fine

![Screenshot_1](https://user-images.githubusercontent.com/15894498/145689765-bc9ea337-a633-4403-808d-c3accd05b7c3.png)

**Comment by Solxanich** [comment]
At any rate, I saw this particular screen as a bug in 1.4 and fixed that to error properly, with a more obvious error;
https://github.com/tModLoader/tModLoader/pull/1812 

Basically, the path breakdown is as follows:
1) Player A starts a server using Steam (lets say it's host&play)
2) Player A invites Player B to join them
3) Player B receives a 16 digit code representing player A's server
4) terraria than attempts to launch a connection from player B to player A using the 16 digit code
5) terraria gets stuck on that screen when the 16 digit code is invalid - ie the host&play session ended from Player A, or they received the wrong code. 

tl;dr - somehow the host&play takes the wrong 16 digit code from Steam, which means this is a steamworks/valve problem. 
An IP server should still work fine for 1.3, but I'm not an expert on it.

**Comment by valikek** [comment]
Use the dedicated server (.exe) and use something like hamachi tunneling, it works for me

**Comment by JavidPack** [comment]
This issue relates to 1.3 tModLoader, which is now legacy and will not receive any further non-critical updates or bug fixes. Please give 1.4 tModLoader a try, there is a good chance this issue might be resolved there already. If you find that 1.4 tModLoader has a similar issue, feel free to open a new issue and fill out the issue template so we can once again track this issue. This issue will now be closed.
